src/arcade/core/util/Graph.java

```diff
@@ -54,96 +57,148 @@ public void clear() {
         nodeToOutBag.clear();
         nodeToInBag.clear();
     }
-    
+
+    /**
+     * Gets all nodes in the graph.
+     *
+     * @return a set containing the nodes
+     */
+    public Set<Node> getAllNodes() {
+        return retrieveNodes();
+    }
+
+    /**
+     * Check to see if a graph contains a node.
+     *
+     * @param node
+     *            the node to add
+     * @return {@code true} if node exists in graph, {@code false} otherwise
+     */
+    public boolean contains(Node node) {
+        return getAllNodes().contains(node);
+    }
+
+    /**
+     * Check to see if a graph contains an edge.
+     *
+     * @param edge
+     *            the edge to add
+     * @return {@code true} if edge exists in graph, {@code false} otherwise
+     */
+    public boolean contains(Edge edge) {
+        return checkEdge(edge);
+    }
+
     /**
      * Gets all edges in the graph.
      *
-     * @return  a bag containing the edges
+     * @return a bag containing the edges
      */
-    public Bag getAllEdges() { return allEdges; }
-    
+    public Bag getAllEdges() {
+        return allEdges;
+    }
+
     /**
      * Gets edges out of the given node.
      *
-     * @param node  the node that edges are from
-     * @return  a bag containing the edges
+     * @param node
+     *            the node that edges are from
+     * @return a bag containing the edges
      */
-    public Bag getEdgesOut(Node node) { return nodeToOutBag.get(node); }
-    
+    public Bag getEdgesOut(Node node) {
+        return nodeToOutBag.get(node);
+    }
+
     /**
      * Gets edges into the given node.
      *
-     * @param node  the node that edges are to
-     * @return  a bag containing the edges
+     * @param node
+     *            the node that edges are to
+     * @return a bag containing the edges
      */
-    public Bag getEdgesIn(Node node) { return nodeToInBag.get(node); }
-    
+    public Bag getEdgesIn(Node node) {
+        return nodeToInBag.get(node);
+    }
+
     /**
      * Gets the in degree at the given node.
      *
-     * @param node  the node
-     * @return  the in degree
+     * @param node
+     *            the node
+     * @return the in degree
      */
     public int getInDegree(Node node) {
         return nodeToInBag.containsKey(node) ? nodeToInBag.get(node).numObjs : 0;
     }
-    
+
     /**
      * Gets the out degree at the given node.
      *
-     * @param node  the node
-     * @return  the out degree
+     * @param node
```

Comment:
These formatting things are maybe going to be addressed by Jessica's gradlew update, but in general it could be good to get those of us using vscode to have a shared java formatting xml so our auto formaters all do the same thing (specifically in this PR looking at the newline vs no newline for javadoc parameter descriptions, and blank spaces (tabs) between functions within a class (or between lines in a function)). I personally have no preference for any format